### PR TITLE
Anchor -- Align icon color to label color when theme.anchor.size.color is specified

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?

To align with previous behavior, icon color should align with label color. With [recent theme enhancement](https://github.com/grommet/grommet/pull/6643), the theme size-specific styling wasn't coming through on icon. Updating iconColor logic to check for color specification.

#### Where should the reviewer start?
src/js/components/Anchor/Anchor.js

#### What testing has been done on this PR?
Tested in storybook, added a jest test.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Related to #6636 

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible.

### CODE REVIEW

File: src/js/components/Anchor/__tests__/Anchor-test.tsx [hunk 1]
```diff
@@ -266,6 +266,11 @@ describe('Anchor', () => {
         </Paragraph>
         <Anchor label="Default anchor with no size prop should receive medium" />
         <Anchor label="Anchor with icon" icon={<svg />} />
+        <Anchor
+          label="Large anchor with icon should receive large color on icon"
```
Comment (ericsoderberghp) [inline src/js/components/Anchor/__tests__/Anchor-test.tsx:270]:
```suggestion
          label="Large anchor with icon should receive color on icon"
```

File: src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap [hunk 2]
```diff
@@ -899,6 +899,22 @@ exports[`Anchor renders size specific theming 1`] = `
...
     </a>
+    <a
+      class="c8"
+    >
+      <span
+        class="c17"
+        style="display: inline-flex;"
+      >
+        <svg
+          color="#7D4CDB"
```
Comment (ericsoderberghp) [inline src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:910]: So, `color` isn't really an `<svg />` attribute. StyledIcon.js converts the `color` property to `stroke` and `fill` in CSS. I'm wondering if we should use an actual grommet-icon to go full circle and make the snapshot looks more real.

Comment (taysea) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:910]: I can do that, will update this test with that.

Comment (taysea) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap]: Done -- you'll see class `c20` applies the appropriate color to the icon.

File: src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap [hunk 3]
```diff
@@ -891,14 +959,48 @@ exports[`Anchor renders size specific theming 1`] = `
...
+      >
+        <svg
+          aria-label="LinkNext"
+          class="c20"
+          viewBox="0 0 24 24"
+        >
+          <path
+            d="M2 12h20m-9-9 9 9-9 9"
+            fill="none"
+            stroke="#000"
```
Comment (ericsoderberghp) [inline src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:994]: shouldn't this be the brand color?

Comment (taysea) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:994]: The way that the color is adjusted is via the class on the surrounding element. the actual stroke always comes through as #000 in the DOM. Class `c20` is applying the correct color via CSS styling.

**Review by ericsoderberghp — APPROVED** [review]

File: src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/Anchor/__tests__/__snapshots__/Anchor-test.tsx.snap:994]: Ah, yes. I used to know that. Thanks for the reminder. :)

**Pull request merged**